test/SILOptimizer/sil_combine_concrete_existential.sil

```diff
@@ -499,3 +499,34 @@ bb0(%0 : $Array<Int>):
   %25 = tuple ()
   return %25 : $()
 }
+
+struct InGuaranteedArgTestNativeObjectWrapper {
+  var i: Builtin.NativeObject
+}
+
+protocol InGuaranteedArgTestProtocol {}
+extension InGuaranteedArgTestNativeObjectWrapper : InGuaranteedArgTestProtocol {}
+
+sil @in_guaranteed_arg_test_callee : $@convention(thin) <τ_0_0 where τ_0_0 : InGuaranteedArgTestProtocol> (@in τ_0_0) -> ()
+
+// Make sure we do not type propagate into the @in function.
+//
```

Comment:
"...The current transformation is not smart enough to remove the destroy_addr. Simply substituting the copy source for the apply argument would introduce a use-after-free."